### PR TITLE
[DOCS] Omit shard failures assertion for incompatible responses 

### PR DESCRIPTION
**Pull request by davidkyle**
The failures in the docs snippet testing seen in #31339 [comment](https://github.com/elastic/elasticsearch/pull/31339#issuecomment-397609887) are due to the framework inserting the assertion `is_false: _shards.failures` which causes errors if the response is not a JSON object. 

The ml datafeed preview API does not return a JSON object and is excluded from the check.

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]
LGTM

**Comment by davidkyle** [comment]
@elasticmachine test this please

**Comment by davidkyle** [comment]
@nik9000 I pushed a second commit updating the regex to exclude API calls on an index such as `index_name/_search` etc.

**Comment by nik9000** [comment]
So now that I look a little more closely, I think we want this for things like searches. So I think the name of the regex is wrong. It should match `/_search`, `/foo/_search`, and `/foo/bar/_search`. Hmmmm. Maybe we should have a list of API that we don't generate this for.... Because so many of them could benefit from this assertion and most of those that don't benefit just have the assertion as a noop.

**Comment by davidkyle** [comment]
And also `_validate`, `_explain`, `_msearch` etc

Yes it is a bit of a pickle and it's tricky to track as there isn't a single class for the response header `BroadcastResponse` and `ReplicationResponse` both write a similar `_shards` object that is a candidate for refactoring. 

I'm afraid we are creating a maintenance burden whitelisting known APIs perhaps we should simply  exclude paths starting with `_cat` or `_xpack`

**Comment by nik9000** [comment]
> I'm afraid we are creating a maintenance burden whitelisting known APIs perhaps we should simply exclude paths starting with _cat or _xpack

I'd expluce `_cat` and the ml API that it doesn't work for and keep it for all of the others.

File: buildSrc/src/test/groovy/org/elasticsearch/gradle/doc/RestTestsFromSnippetsTaskTest.groovy [hunk 1]
```diff
@@ -47,4 +45,10 @@ class RestTestFromSnippetsTaskTest extends GroovyTestCase {
         assertEquals("\"foo\": \"bort\\n baz\"",
             replaceBlockQuote("\"foo\": \"\"\"bort\n baz\"\"\""));
     }
+
+    void testIsDocWriteRequest() {
```
Comment (nik9000) [inline buildSrc/src/test/groovy/org/elasticsearch/gradle/doc/RestTestsFromSnippetsTaskTest.groovy]: Can you change the name of the test?

**Review by nik9000 — APPROVED** [review]
Please change the name of the test and merge when ready!

**Pull request merged**